lib/src/components/NxTooltip/__tests__/NxTooltip.test.tsx

```diff
@@ -4,113 +4,278 @@
+  it('adds specified classes to an element within the tooltip and not to the child', async function() {
+    const user = userEvent.setup(),
+        view = quickRender({ className: 'foo' }),
+        defaultView = quickRender(),
+        child = view.getByTestId('foo'),
+        defaultChild = defaultView.getByTestId('foo');
 
-    expect(component).toMatchSelector(Tooltip);
-    expect(component).toHaveProp('children', minimalProps.children);
-    expect(component).toHaveProp('open', false);
-    expect(component).toHaveProp('onOpen', onOpen);
-    expect(component).toHaveProp('onClose', onClose);
-  });
+    await user.hover(child);
+    const tooltip = await screen.findByRole('tooltip'),
+        elWithClassName = tooltip.querySelector('.foo')!,
+        otherClasses = Array.from(elWithClassName.classList).filter(c => c !== 'foo');
 
-  it('replaces null title with empty string, and all other null props with undefined', function() {
-    const component = getTooltip({
-      title: null,
-      className: null,
-      open: null,
-      onOpen: null,
-      onClose: null,
-      placement: null
-    });
+    expect(elWithClassName).toBeInTheDocument();
 
-    expect(component).toHaveProp('open', undefined);
-    expect(component).toHaveProp('onOpen', undefined);
-    expect(component).toHaveProp('onClose', undefined);
-    expect(component).toHaveProp('title', '');
-  });
+    await user.hover(defaultChild);
+    const defaultTooltip = await screen.findByRole('tooltip');
 
-  it('passes through string titles after an asynchronous initialization period', function() {
-    const emptyComponent = getMounted({ title: '' });
-    act(() => { jest.runAllTimers(); });
-    emptyComponent.update();
-    expect(emptyComponent.children()).toHaveProp('title', '');
+    for (const cls of otherClasses) {
+      expect(defaultTooltip.querySelector(`.${cls}`)).toBeInTheDocument();
+    }
 
-    const component = getMounted({ title: 'foo' });
-    act(() => { jest.runAllTimers(); });
-    component.update();
-    expect(component.children()).toHaveProp('title', 'foo');
+    expect(child.querySelector('.foo')).not.toBeInTheDocument();
   });
 
-  it('passes through JSX titles', function() {
-    const title = <div className="foo" />,
-        component = getMounted({ title });
+  it('renders the tooltip open automatically if open is true', async function() {
+    quickRender({ open: true });
 
-    act(() => { jest.runAllTimers(); });
-    component.update();
+    const tooltip = await screen.findByRole('tooltip');
 
-    expect(component.children()).toHaveProp('title', title);
+    expect(tooltip).toBeInTheDocument();
+    expect(tooltip).toHaveTextContent('tip');
   });
 
-  it('passes "nx-tooltip" as the `tooltip` property on the `classes` prop of the Tooltip', function() {
-    expect(getTooltip().prop('classes').tooltip).toBe('nx-tooltip');
-  });
+  it('calls onOpen when the tooltip renders by hover but not due to the open prop', async function() {
```

Comment:
When I tested the second quickRender function with different props (ie. `quickRender({ onOpen, className: 'foo', placement: 'bottom'})` ), and used `await user.hover(child)` for the second time, the test still came back as having `onOpen` being called one time. Perhaps the `open` prop set to `true` should be tested first to show `onOpen` wasn't called, and then use `quickRender({onOpen})` & `await user.hover(child)` to assert `onOpen` has indeed been called.

Reply:
>When I change the code as follows, it passes

Thanks for writing out a solution. I called `hover` and `waitFor`, but completely forgot to assign a variable to the second `quickRender` (just called `quickRender({ onOpen, placement: 'bottom'}), which of course caused things to fail. Thanks!